### PR TITLE
Pass MacroBaseConf to each OutlierDetector

### PR DESCRIPTION
**Pull request by viveksjain**
Implement #83. BTW, it's also a little strange to me that MacroBaseConf needs to know how to construct DataLoaderType, DetectorType, BandwidthAlgorithm etc. But this is enough refactoring for today. :)

### CODE REVIEW

**Comment by viveksjain** [comment]
Gah I broke all the tests.

**Comment by viveksjain** [comment]
mcdAlpha and mcdStoppingDelta from the conf didn't seem to be actually used earlier. Adding it in broke one of the tests (which I fixed) but I hope it doesn't break a previous workflow.

File: src/test/java/macrobase/analysis/outlier/MovingAverageTest.java [hunk 1]
```diff
@@ -4,15 +4,19 @@
...
 import org.apache.commons.math3.linear.ArrayRealVector;
 import org.junit.Test;
 
+import com.google.common.collect.ImmutableMap;
+
 public class MovingAverageTest {
     @Test
     public void simpleTest() {
-        MovingAverage ma = new MovingAverage(3);
+        MacroBaseConf conf = new MacroBaseConf(ImmutableMap.of(MacroBaseConf.TUPLE_WINDOW, 3));
```
Comment (pbailis) [inline src/test/java/macrobase/analysis/outlier/MovingAverageTest.java]: Slightly prefer a `new MacroBaseConf().set(MacroBaseConf.TUPLE_WINDOW, 3)`. May require making `MacroBase.set` return `this`.

**Comment by pbailis** [comment]
:100: Had one comment on style, that's unrelated.

**Comment by viveksjain** [comment]
Agreed, I do like that style better.

**Comment by pbailis** [comment]
An even fancier way is to add a static constructor:

`MacroBaseConf.create().set()...`

`public static MacroBaseConf create() { return new MacroBaseConf(); }`

**Comment by viveksjain** [comment]
Hmm that's one of those "design patterns," isn't it *wrinkles nose in disgust*? Not that there's anything wrong with that, but I don't see the point in this case. If the static constructor was used to create a builder that is then immutable after being built, I understand why you would want to have it, but if it's just calling new then that's 2 ways of doing the same thing.

**Comment by pbailis** [comment]
Fair -- let's merge this!

**Pull request merged**